modules/tappxBidAdapter.js

```diff
@@ -0,0 +1,284 @@
+'use strict';
+
+import * as utils from '../src/utils.js';
+import { registerBidder } from '../src/adapters/bidderFactory.js';
+import { BANNER } from '../src/mediaTypes.js';
+import { config } from '../src/config.js';
+
+const BIDDER_CODE = 'tappx';
+var HOST;
+var hostDomain;
+
+export const spec = {
+  code: BIDDER_CODE,
+  supportedMediaTypes: [BANNER],
+
+  /**
+   * Determines whether or not the given bid request is valid.
+   *
+   * @param {BidRequest} bid The bid params to validate.
+   * @return boolean True if this is a valid bid, and false otherwise.
+  */
+  isBidRequestValid: function(bid) {
+    if ((bid.params == null) || (bid.params.endpoint == null) || (bid.params.tappxkey == null)) {
+      utils.logWarn(`[TAPPX]: The Tappx endpoint and Tappxkey are mandatory. ${JSON.stringify(bid)}`);
+      return false;
+    }
+    return true;
+  },
+
+  /**
+   * Takes an array of valid bid requests, all of which are guaranteed to have passed the isBidRequestValid() test.
+   * Make a server request from the list of BidRequests.
+   *
+   * @param {validBidRequests[]} - an array of bids
+   * @return ServerRequest Info describing the request to the server.
+   */
+  buildRequests: function(validBidRequests, bidderRequest) {
+    // Host definition
+    HOST = getParameters(validBidRequests, 'params.host');
+    hostDomain = HOST.split('/', 1)[0];
+
+    const ENDPOINT = getParameters(validBidRequests, 'params.endpoint');
+    const TAPPXKEY = getParameters(validBidRequests, 'params.tappxkey');
+    const BIDFLOOR = getParameters(validBidRequests, 'params.bidfloor');
+    const bannerMediaType = getParameters(validBidRequests, 'mediaTypes.banner');
+
+    let requests = [];
+    let payload = {};
+    let publisher = {};
+    let tagid;
+    let api = {};
+
+    // > App/Site object
+    if (getParameters(validBidRequests, 'params.app')) {
+      let app = {};
+      app.name = getParameters(validBidRequests, 'params.app.name');
+      app.bundle = getParameters(validBidRequests, 'params.app.bundle');
+      app.domain = getParameters(validBidRequests, 'params.app.domain');
+      publisher.name = getParameters(validBidRequests, 'params.app.publisher.name');
+      publisher.domain = getParameters(validBidRequests, 'params.app.publisher.domain');
+      tagid = `${app.name}_typeAdBanVid_${getOs()}`;
+      payload.app = app;
+      api[0] = getParameters(validBidRequests, 'params.api') ? getParameters(validBidRequests, 'params.api') : [3, 5];
+    } else {
+      let site = {};
+      site.name = window.location.hostname;
```

Comment:
are you considering the case when PBJS is running inside an iframe? Did you check what is provided by `refererInfo` inside `bidRequest` and if you can use that instead?

Reply:
Following your advice, we have updated the adapter to fix this mistake in the commit [31d23ee](https://github.com/prebidtappx/Prebid.js/commit/31d23ee4a7e5f2e5973a1d59c264d87aeb4b9887)